pandera/schemas.py

```diff
@@ -271,60 +272,42 @@ def get_dtype(self, dataframe: pd.DataFrame) -> Dict[str, str]:
         }
 
     @property
-    def pandas_dtype(
+    def dtype(
         self,
-    ) -> Union[str, dtypes.PandasDtype, dtypes.PandasExtensionType]:
-        """Get the pandas dtype property."""
-        return self._pandas_dtype
+    ) -> PandasDtypeInputTypes:
+        """Get the dtype property."""
+        return self._dtype
```

Comment:
consider returning `pandas_engine.Engine.dtype(self._dtype)` and the setter can just `self._dtype = value`. This way `_dtype` preserves the raw user-provided value but the public property is whatever the pandas engine resolves to

Reply:
This change would multiply the number of type conversions.